images/bootstrap/pkg/service/cassandra.go

```diff
@@ -0,0 +1,169 @@
+package service
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	namespace      string
+	pod            string
+	configmap      string
+	ipAddress      string
+	bootstrapWait  string
+	annotationLock = "cassandra.kudo.dev/annotationLock"
+	replaceFile    = "/var/lib/cassandra/replace.ip"
+)
+
+type CassandraService struct {
+	CMService *ConfigMapLock
+}
+
+func NewCassandraService(client *kubernetes.Clientset) *CassandraService {
+	return &CassandraService{
+		CMService: &ConfigMapLock{client},
+	}
+}
+
+func (c *CassandraService) SetReplaceIP() (bool, error) {
```

Comment:
I think this func should contain at least some sort of retry-functionality. I don't think it will happen too often, but it will happen that two nodes try to acquire/write to the config map at the same time and one of them will fail at some point.

Reply:
the issue here is we probably want it to fail early. Because this is bootstrap binary which runs in background. If we don't acquire its better to do an exit(1) and restart. 
And we cannot make Cassandra process wait either as we need `nodetool status` to see if replaced IP is `UN/UJ` also. Let's sync about this on how we could probably improve this without any perf issues.